providers/dns/hetzner/internal/client.go

```diff
@@ -129,7 +133,7 @@ func (c *Client) DeleteRecord(recordID string) error {
 
 // GetZoneID gets the zone ID for a domain.
 func (c *Client) GetZoneID(domain string) (string, error) {
-	zones, err := c.getZones()
+	zones, err := c.getZones(domain, &Pagination{Page: 1, PerPage: pageSize})
```

Comment:
This is a curious one. I was wondering, why the page number isn't iterated on.

The docs state for the `name` parameter (here given as `domain`), that it "will return an array with one or no results". Also, there's a curiously worded "400 Pagination selectors are mutually exclusive" response (without further explanation).

I'm interpreting that as applying pagination (`page=1&per_page=100`) would be either unnecessary (because there's only one page with 0-1 entries), or erroneous (because we've selected to paginate by name, not by cursor).